plugins/src/main/java/com/google/fhir/gateway/plugin/PatientAccessChecker.java

```diff
@@ -119,17 +119,21 @@ private AccessDecision processPost(RequestDetailsReader requestDetails) {
     return processCreate(requestDetails);
   }
 
+  private Boolean validatePatientIds(Set<String> patientIds) {
+    return patientIds.size() == 1 && authorizedPatientId.equals(patientIds.iterator().next());
```

Comment:
So this is related to my other comment about `findPatientsFromParams` contract: If its returned value is not guaranteed to be non-null, then I think we potentially have an NPE issue here (Java's null handling is really brutal TBH). So better if we guarantee it is non-null and add it to its javadoc contract.